lib/soap.d.ts

```diff
@@ -159,7 +159,7 @@ export function listen(server: any, options: IServerOptions): Server;
 
 export class HttpClient {
     constructor(options?: IOptions);
-    buildRequest(rurl: string, data: any | string, exheaders?: { [key: string]: any }, exoptions?: { [key: string]: any }): any;
+    buildRequest(rurl: string, data: any | string, exheaders: { [key: string]: any }, exoptions: { [key: string]: any }): any;
```

Comment:
Sync ts with jsdoc

Reply:
I'm a bit confused.  They weren't mandatory before.  Why should they be mandatory now?

Reply:
Why not. The jsdoc does not mark them as optional, and there is only one instance, which does not set them to an empty object like all the others. So if we just set them to an empty object by default in this one instance, we can make the parameters mandatory and do not have to handle a falsy value in this function, thus taking complexity out of it

Reply:
Documentation is one thing.  If the implementation had them as optional in the past, then I'm of the opinion that we should update the documentation i.e. ts file or jsdoc.

Reply:
Is this even a method that most users of `node-soap` would use?  If not, then I'm fine with this change.